### PR TITLE
Dev with route and linting fix

### PR DESCRIPTION
**Pull request by sofiiagran**
* Now everything is linted and routes follow microservice guidelines.
* Commented out some stuff:
       * commented out leaderboard.js, since it is not used and gives linting errors
       * commented out function ensureNewValues in validation.js since it gives linting error and is not in use
* The route updates is also updated on mobile

IMPORTANT: test with dev_withRouteFix on mobile, so the routes match.

### CODE REVIEW

**Comment by sofiiagran** [comment]
@JackSMunn @Jatewo @MortenMunk @TheMOHD3886 @MaizeHH @Havkost 

Ved I måske hvorfor jeres tests failer?

**Comment by Jatewo** [comment]
> @JackSMunn @Jatewo @MortenMunk @TheMOHD3886 @MaizeHH @Havkost
> 
> Ved I måske hvorfor jeres tests failer?

Har du tjekket loggen?

**Comment by sofiiagran** [comment]
@Jatewo Ja, den giver 400 når i forventer 200

**Comment by Jatewo** [comment]
> @Jatewo Ja, den giver 400 når i forventer 200

Så er der jo lavet breaking changes. Det er ikke vores skyld, at vores tests fejler, hvis testene har bestået før. Så er det jo nogle af ændringerne på branchen, der er skyld i det.

File: .eslintrc.js [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
+    }
+  ],
+  'parserOptions': {
+    'ecmaVersion': 'latest'
+  },
+  'ignorePatterns': [
+    '**/*.test.js', '**/*.spec.js', '**/*.config.js', 'jest-setup.js', 'db.js'
+  ],
+  'rules': {
+    'indent': ['error', 2],
```
Comment (Jatewo) [inline .eslintrc.js:29]: This should be 'tab' - not 2

File: .eslintrc.js [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
-	],
-	'parserOptions': {
-		'ecmaVersion': 'latest'
-	},
-	'rules': {
-		'indent': [
-			'error',
-			'tab',
-		],
-		'linebreak-style': [
```
Comment (Jatewo) [inline .eslintrc.js:30]: I think we could just keep this - unless you have a good reason for removing it

File: .github/workflows/Linting.yml [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
+      - name: Check out code
         uses: actions/checkout@v2
 
-      - name: Use Node.js 16.x
-        uses: actions/setup-node@v1
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
         with:
-          node-version: 16.x
+          node-version: 14
```
Comment (Jatewo) [inline .github/workflows/Linting.yml:19]: Why downgrade node-version?

File: .github/workflows/Linting.yml [hunk 4]
```diff
@@ -1,38 +1,28 @@
...
+      
+      - name: List directories and files
```
Comment (Jatewo) [inline .github/workflows/Linting.yml:21]: Why are we running this command?

File: __tests__/routes/studentRoutes.test.js [hunk 5]
```diff
@@ -744,6 +744,8 @@ describe('PATCH /api/students/:userId/complete', () => {
 	});
 });
 
+/*  Commented out as the route is not in use
```
Comment (Jatewo) [inline __tests__/routes/studentRoutes.test.js:747]: Maybe delete?

File: helpers/leaderboard.js [hunk 6]
```diff
@@ -1,3 +1,7 @@
+/*
+ * This gives linting error, and since it is not used, it is commented out.
+ *
```
Comment (Jatewo) [inline helpers/leaderboard.js:3]: Delete?

File: package.json [hunk 7]
```diff
@@ -18,7 +18,7 @@
     "docker_down": "docker-compose down",
     "docker_build": "docker-compose up -d --build",
     "docker_clean_db": "docker-compose down && docker volume rm educado-backend_mongodb_data && docker-compose up -d --build mongodb",
-    "linting": "eslint --ext .js,.jsx,.ts,.tsx . --fix",
+    "lint": "eslint --fix .",
```
Comment (Jatewo) [inline package.json:21]: Should we force the --fix for everyone?

**Review by Jatewo — CHANGES_REQUESTED** [review]
I have left a few comments with some requested changes. Please have a look 👍

File: .eslintrc.js (hunk 1 above)
Comment (sofiiagran) [reply .eslintrc.js:29]: tab does not work, we use this format for all repos and all workflows. It will not be changed.

File: .eslintrc.js (hunk 2 above)
Comment (sofiiagran) [reply .eslintrc.js:30]: Yes I have a good reason. I changed it to match the one we have on educado-mobile. The one we used on the backend did not work, while the one used on mobile worked. Therefore, we keep the one that does work :))

File: .github/workflows/Linting.yml (hunk 3 above)
Comment (sofiiagran) [reply .github/workflows/Linting.yml:19]: mistake

File: package.json (hunk 7 above)
Comment (sofiiagran) [reply package.json:21]: yes

**Review by loui981c — APPROVED** [review]
Looks goodie, tested the backend with mobile and works as expected

File: .eslintrc.js (hunk 1 above)
Comment (Jatewo) [reply .eslintrc.js:29]: What do you mean 'tab does not work'? It has worked right up until now. And no we shouldn't be using 2 spaced in any other repos, if we do, that's a mistake

File: .eslintrc.js (hunk 2 above)
Comment (Jatewo) [reply .eslintrc.js:30]: I don't know if it's important, but unless you have found out that this is what is wrong with a file that does not work, then this is hardly a good reason to delete it...

File: .github/workflows/Linting.yml (hunk 4 above)
Comment (Jatewo) [reply .github/workflows/Linting.yml]: If you want to you can easily run it in your terminal. My question is to why we would run this in the GitHub Action

File: package.json (hunk 7 above)
Comment (Jatewo) [reply package.json]: Why?

Comment (sofiiagran) [reply package.json]: @Jatewo It is fixed, now there are two seperate, one with fix and one without

File: helpers/leaderboard.js (hunk 6 above)
Comment (sofiiagran) [reply helpers/leaderboard.js:3]: We will keep it for next year and explain why in README

File: .github/workflows/Linting.yml (hunk 4 above)
Comment (sofiiagran) [reply .github/workflows/Linting.yml]: @Jatewo i removed it before merging :)))

File: .eslintrc.js (hunk 1 above)
Comment (sofiiagran) [reply .eslintrc.js:29]: @Jatewo please look at the workflow in educado-mobile

**Pull request merged**